src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -1040,11 +1040,25 @@ public final HttpClient metrics(boolean metricsEnabled) {
 	 * @param metricsEnabled if true enables the metrics on the server.
 	 * @param recorder the {@link HttpClientMetricsRecorder}
 	 * @return a new {@link HttpClient}
+	 * @deprecated  as of 0.9.7. Use {@link #metrics(boolean, Supplier)}
 	 */
+	@Deprecated
 	public final HttpClient metrics(boolean metricsEnabled, HttpClientMetricsRecorder recorder) {
 		return tcpConfiguration(tcpClient -> tcpClient.metrics(metricsEnabled, recorder));
 	}
 
+	/**
+	 * Specifies whether the metrics are enabled on the {@link HttpClient}.
+	 * All generated metrics are provided to the specified recorder.
```

Comment:
nitpick: could add `, which is only instantiated if metrics are being enabled`

Reply:
updated